### ISSUE TITLE
PrepareResourcesDone 

### ISSUE THREAD

**Issue by tomwiel** [castle-engine/castle-engine#156]
I have no testcase, but it looks like viewport.prepareResourcesDone could need reset after loading different scene?

**Comment by michaliskambi** [comment]
That is true. Good point! I'll think how to address it best -- we could just watch for adding/removing scenes.

Note that nothing bad will happen anyway (if `PrepareResources` will not execute from `BeforeRender`), it's just that some resources will then be prepared during `Render` and will cause temporary FPS drop due to something done just once.
